targets/frontend/src/components/storage/DropZone.js

```diff
@@ -0,0 +1,58 @@
+/** @jsx jsx */
+
+import PropTypes from "prop-types";
+import { useCallback, useState } from "react";
+import { useDropzone } from "react-dropzone";
+import { mutate } from "swr";
+import { jsx, Spinner } from "theme-ui";
+
+const defaultStyles = {
+  border: "2px dotted silver",
+  borderRadius: "small",
+  p: "medium",
+};
+
+export function DropZone({ onDrop: onDropCallback, customStyles }) {
+  const [uploading, setUploading] = useState(false);
+  const onDrop = useCallback(
+    async (acceptedFiles) => {
+      setUploading(true);
+      const formData = new FormData();
+      for (const i in acceptedFiles) {
+        if (acceptedFiles[i] instanceof File) {
+          console.log(acceptedFiles[i].path);
+          console.log(acceptedFiles[i]);
+          formData.append(acceptedFiles[i].path, acceptedFiles[i]);
+        }
+      }
+      onDropCallback(formData).finally(() => {
+        setUploading(false);
+        mutate("files");
```

Comment:
I propose to call mutate("files") in the drop callback we pass, to let the upper component to handle swr

Reply:
Yep it's gonna handle the loading state too. Having a forcer promise callback is too clumsy